From 49f56111b8f2cc83d95d333f8803a3f1c5c51a1c Mon Sep 17 00:00:00 2001
From: Chris Nisbet <nischris@gmail.com>
Date: Wed, 12 Feb 2020 21:00:31 +1300
Subject: [PATCH] blobmsg: fix wrong payload len passed from
 blobmsg_check_array

Fix incorrect use of blobmsg_len() on passed blobmsg to
blobmsg_check_array_len()  introduced in commit 379cd33d1992 ("fix wrong payload len passed from blobmsg_check_array) by using correct
blob_len().

By using blobmsg_len() a value too small was passed to blobmsg_check_array() which could lead to this function returning an error when there is none.

Signed-off-by: Chris Nisbet <nischris@gmail.com>
---
 blobmsg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/blobmsg.c b/blobmsg.c
index 7620784..8b9877d 100644
--- a/blobmsg.c
+++ b/blobmsg.c
@@ -114,7 +114,7 @@ bool blobmsg_check_attr_len(const struct blob_attr *attr, bool name, size_t len)
 
 int blobmsg_check_array(const struct blob_attr *attr, int type)
 {
-	return blobmsg_check_array_len(attr, type, blobmsg_len(attr));
+	return blobmsg_check_array_len(attr, type, blob_len(attr));
 }
 
 int blobmsg_check_array_len(const struct blob_attr *attr, int type, size_t len)
-- 
2.20.1

